rosa-hypershift/rosa-hosted-wrapper.py

```diff
@@ -577,6 +782,99 @@ def _cleanup_cluster(rosa_cmnd, cluster_name, mgmt_cluster_name, my_path, my_uui
         metadata["timestamp"] = datetime.datetime.utcnow().isoformat()
         es_ignored_metadata = ""
         common._index_result(es, index, metadata, es_ignored_metadata, index_retry)
+    if args.manually_cleanup_aws_iam:
+        aws_roles = _destroy_aws_iam_roles(cluster_name)
+        if aws_roles != 0:
+            logging.error("Failed to destroy AWS IAM Roles of %s (%s)" % (cluster_name, metadata['cluster_id']))
+        aws_oidc = _destroy_aws_iam_oidc(cluster_name, metadata['cluster_id'])
+        if aws_oidc != 0:
+            logging.error("Failed to destroy AWS IAM OIDC of %s (%s)" % (cluster_name, metadata['cluster_id']))
+
+
+def _destroy_aws_iam_oidc(cluster_name, cluster_id):
+    logging.info("Geting AWS OpenID Providers of %s" % cluster_name)
+    oidc_list_cmnd = ["aws", "iam", "list-open-id-connect-providers", "--output", "json"]
+    logging.debug(oidc_list_cmnd)
+    oidc_list_process = subprocess.Popen(oidc_list_cmnd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    oidc_list_stdout, oidc_list_stderr = oidc_list_process.communicate()
+    return_code = 0
+    if oidc_list_process.returncode == 0:
+        for provider in json.loads(oidc_list_stdout.decode("utf-8"))['OpenIDConnectProviderList']:
+            if cluster_id in provider['Arn']:
+                logging.info("Deleting OIDC Provider %s of cluster: %s" % (provider['Arn'], cluster_name))
+                delete_oidc_cmnd = ["aws", "iam", "delete-open-id-connect-provider", "--open-id-connect-provider-arn", provider['Arn']]
+                logging.debug(delete_oidc_cmnd)
+                delete_oidc_process = subprocess.Popen(delete_oidc_cmnd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+                delete_oidc_stdout, delete_oidc_stderr = delete_oidc_process.communicate()
+                if delete_oidc_process.returncode != 0:
+                    logging.error("Failed to delete OIDC provider %s for cluster %s" % (provider['Arn'], cluster_name))
```

Comment:
No retry required on AWS IDP and role deletes ?

Reply:
I havent seen any error on those commands that will be fixed on a retry. if fails it is not a matter of waiting for another try

Reply:
okay, then its fine.